packages/wrangler/src/sentry/index.ts

```diff
@@ -0,0 +1,132 @@
+import * as Sentry from "@sentry/node";
+import { rejectedSyncPromise } from "@sentry/utils";
+import { fetch } from "undici";
+import { version as wranglerVersion } from "../../package.json";
+import { confirm } from "../dialogs";
+import { logger } from "../logger";
+import type { BaseTransportOptions, TransportRequest } from "@sentry/types";
+import type { RequestInit } from "undici";
+
+let sentryReportingAllowed = false;
+
+// The SENTRY_DSN is provided at esbuild time as a `define` for production and beta releases.
+// Otherwise it is left undefined, which disables reporting.
+declare const SENTRY_DSN: string;
+
+/* Returns a Sentry transport for the Sentry proxy Worker. */
+export const makeSentry10Transport = (options: BaseTransportOptions) => {
+	let eventQueue: [string, RequestInit][] = [];
+
+	const transportSentry10 = async (request: TransportRequest) => {
+		/* Adds helpful properties to the request body before we send it to our
+    proxy Worker. These properties can be parsed out from the NDJSON in
```

Comment:
NDJSON?

Reply:
That's how Sentry sends events internally, as (N)ewline (D)elimited JSON